GitUI/CommandsDialogs/SettingsDialog/Pages/FormBrowseRepoSettingsPage.cs

```diff
@@ -1,78 +1,502 @@
+            }
+        }
+
+        private sealed class ResourceImagesTypeEditor : UITypeEditor
+        {
+            public override UITypeEditorEditStyle GetEditStyle(ITypeDescriptorContext context)
+            {
+                return UITypeEditorEditStyle.DropDown;
+            }
+
+            public override bool GetPaintValueSupported(ITypeDescriptorContext context)
+            {
+                return true;
+            }
+
+            public override void PaintValue(PaintValueEventArgs e)
+            {
+                var imageKey = e.Value?.ToString();
+
+                if (imageKey != null)
+                {
+                    var rectangle = e.Bounds;
+
+                    rectangle.Inflate(-2, -1);
+
+                    e.Graphics.DrawImage(ResourceImagesProvider.Images[imageKey], rectangle);
+                }
+            }
+
+            public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
+            {
+                var editorService = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
+
+                if (editorService is null)
+                {
+                    return value;
+                }
+
+                var listView = new ListView
+                {
+                    Scrollable = true,
+                    View = View.List,
+                    MultiSelect = false,
+                    BorderStyle = BorderStyle.None
+                };
+
+                var imageList = new ImageList
+                {
+                    ColorDepth = ColorDepth.Depth32Bit
+                };
+
+                listView.SmallImageList = imageList;
+                listView.LargeImageList = imageList;
+
+                foreach (var image in ResourceImagesProvider.Images)
+                {
+                    imageList.Images.Add(image.Key, image.Value);
+
+                    listView.Items.Add(image.Key, image.Key);
+                }
+
+                listView.Click += (object sender, EventArgs e) =>
+                {
+                    editorService.CloseDropDown();
+                };
+
+                editorService.DropDownControl(listView);
+
+                if (listView.SelectedItems.Count > 0)
+                {
+                    return listView.SelectedItems[0].Text;
+                }
+
+                return value;
+            }
         }
 
-        private void cboTerminal_Enter(object sender, EventArgs e)
+        [TypeConverter(typeof(PropertySorter))]
+        private sealed class ShellViewModel : IShell, INotifyPropertyChanged, ICheckable
         {
-            _cboTerminalPreviousIndex = cboTerminal.SelectedIndex;
+            private const string ShellCategory = "Shell";
+
+            public static ShellViewModel Create()
+            {
+                return new ShellViewModel();
+            }
+
+            public static ShellViewModel Create(IShell shell)
```

Comment:
Use `implicit operator` instead

Reply:
User-defined conversions are not allowed to convert from or to interface-types. In particular, this restriction ensures that no user-defined transformations occur when converting to an interface-type, and that a conversion to an interface-type succeeds only if the object being converted actually implements the specified interface-type.